packages/video_player/video_player/example/pubspec.yaml

```diff
@@ -14,6 +14,15 @@ dependencies:
     # the parent directory to use the current plugin's version.
     path: ../
 
+# For development / testing purpose only.
+# When using video_player in a real application
+# you won't directly depend on video_player_platform_interface at all.
+# The example app is bundled with the plugin so we use a path dependency on
+# the parent directory to use the current plugin's version.
+dependency_overrides:
+  video_player_platform_interface:
+    path: ../../video_player_platform_interface
```

Comment:
This will need to be removed, and the PR split into two parts, before landing.

Reply:
Doesn't it make sense to keep the override to make anyone cloning the source to work with local copy?

Just like `video_player:` dependency in the example is declared as `path: ../` to point to local copy of `video_player` package, this would do the same for `video_player_platform_interface` package

Reply:
> Doesn't it make sense to keep the override to make anyone cloning the source to work with local copy?

There are several issues with that, most notably that it causes the integration tests to run against a configuration that is different from what actual clients of this package would run, making it easy to have a PR pass CI and be landed and published, but break everyone because it doesn't work with the published interface package.

> Just like `video_player:` dependency in the example is declared as `path: ../` to point to local copy of `video_player` package

The example is *part* of that package. It's an internal reference.

> this would do the same for `video_player_platform_interface` package

Overriding dependencies in a multi-package graph is definitely not the same thing.

Reply:
OK, thanks, I understand the problem now. May I keep it commented out or should completely delete?

Reply:
We don't check in commented out code, generally speaking. It's very easy for someone who wants this behavior for temporary testing to re-add it.

Reply:
OK, I'll remove the override. But the https://github.com/flutter/plugins/pull/3702 need to be landed first (if there aren't any other issues in the current PR) because the example won't built otherwise

Reply:
Deleted it now